core/server/proxy/src/main/java/alluxio/proxy/s3/RateLimitInputStream.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.proxy.s3;
+
+import com.google.common.util.concurrent.RateLimiter;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * This class is a wrapper for InputStream which limit rate when reading bytes.
+ */
+public class RateLimitInputStream extends InputStream {
+
+  private final InputStream mInputStream;
+  private final RateLimiter mRateLimiter;
```

Comment:
so this is to limit one http connection's reading is it?  why not on a global level throttling? to control, coz as client num(num of read http conn increases, the rate is still unbounded if ratelimiter is for individual read?) i'm assuming there's barely writes, therefore the full bandwidth is consumed by reads?

Reply:
We are limiting the speed just to protect the client side, while on the server side we want it to reach maximum performance, but in terms of functional integrity, the server side should do a global speed limit, which I will finish later.

Reply:
In our scenario, the client is on k8s and is single-threaded. The single connection speed limit is done to prevent a single pod from overloading the k8s node network bandwidth and affecting other pods. Maybe I should not just consider this scenario, there are other scenarios that require a global speed limit.